src/err/err_state.rs

```diff
@@ -321,3 +353,79 @@ fn raise_lazy(py: Python<'_>, lazy: Box<PyErrStateLazyFn>) {
         }
     }
 }
+
+#[cfg(test)]
+mod tests {
+    use std::sync::OnceLock;
+
+    use crate::{exceptions::PyValueError, PyErr, PyErrArguments, PyObject, Python};
+
+    #[test]
+    #[should_panic(expected = "Re-entrant normalization of PyErrState detected")]
+    fn test_reentrant_normalization() {
+        static ERR: OnceLock<PyErr> = OnceLock::new();
+
+        struct RecursiveArgs;
+
+        impl PyErrArguments for RecursiveArgs {
+            fn arguments(self, py: Python<'_>) -> PyObject {
+                // .value(py) triggers normalization
+                ERR.get()
+                    .expect("is set just below")
+                    .value(py)
+                    .clone()
+                    .into()
+            }
+        }
+
+        ERR.set(PyValueError::new_err(RecursiveArgs)).unwrap();
+
+        Python::with_gil(|py| {
+            ERR.get().expect("is set just above").value(py);
+        })
+    }
+
+    #[test]
+    fn test_no_deadlock_thread_switch() {
+        static ERR: OnceLock<PyErr> = OnceLock::new();
```

Comment:
More of a style thing I guess, but I think you can avoid the static if you use `std::thread::scope`

Reply:
I think in this case I can't, because I use `Err` inside a trait implementation.